diff/walking/differ.go

```diff
@@ -111,27 +126,32 @@ func (s *walkingDiff) Compare(ctx context.Context, lower, upper []mount.Mount, o
 				}
 			}
 
+			var st archive.DiffStat
+			onWriteDiffCompletion := func(v archive.DiffStat) {
+				st = v
+			}
+			diffOpts := []archive.DiffOpt{archive.WithOnWriteDiffCompletion(onWriteDiffCompletion)}
 			if isCompressed {
 				dgstr := digest.SHA256.Digester()
 				compressed, err := compression.CompressStream(cw, compression.Gzip)
 				if err != nil {
 					return errors.Wrap(err, "failed to get compressed stream")
 				}
-				err = archive.WriteDiff(ctx, io.MultiWriter(compressed, dgstr.Hash()), lowerRoot, upperRoot)
+				err = archive.WriteDiff(ctx, io.MultiWriter(compressed, dgstr.Hash()), lowerRoot, upperRoot, diffOpts...)
 				compressed.Close()
 				if err != nil {
 					return errors.Wrap(err, "failed to write compressed diff")
 				}
 
-				if config.Labels == nil {
-					config.Labels = map[string]string{}
-				}
-				config.Labels[uncompressed] = dgstr.Digest().String()
+				config.Labels[LabelUncompressed] = dgstr.Digest().String()
 			} else {
-				if err = archive.WriteDiff(ctx, cw, lowerRoot, upperRoot); err != nil {
+				if err = archive.WriteDiff(ctx, cw, lowerRoot, upperRoot, diffOpts...); err != nil {
 					return errors.Wrap(err, "failed to write diff")
 				}
 			}
+			if st.Empty {
+				config.Labels[LabelEmptyLayer] = ""
+			}
 
 			var commitopts []content.Opt
 			if config.Labels != nil {
```

Comment:
since we have initialized the `config.Labels` at the beginning, I think we should check the length before appending the option. what do you think?

Reply:
if there is no harm to add empty labels, I am ok to remove the `if` condition.

Reply:
updated